public_html/wp-content/themes/wporg-events-2023/src/event-list/index.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Block Name: WordPress Event List
+ * Description: List of WordPress Events.
+ *
+ * @package wporg
+ */
+
+namespace WordPressdotorg\Theme\Events_2023\WordPress_Event_List;
+
+use function WordPressdotorg\MU_Plugins\Google_Map\{get_events};
+
+add_action( 'init', __NAMESPACE__ . '\init' );
+
+/**
+ * Registers the block using the metadata loaded from the `block.json` file.
+ * Behind the scenes, it registers also all assets so they can be enqueued
+ * through the block editor in the corresponding context.
+ *
+ * @see https://developer.wordpress.org/reference/functions/register_block_type/
+ */
+function init() {
+	register_block_type(
+		dirname( dirname( __DIR__ ) ) . '/build/event-list',
+		array(
+			'render_callback' => __NAMESPACE__ . '\render',
+		)
+	);
+}
+
+/**
+ * Render the block content.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the block markup.
+ */
+function render( $attributes, $content, $block ) {
+	$facets = array(
+		'search' => get_query_var( 's' ) ?? '',
+		'type'   => get_query_var( 'event_type' ) ?? '',
+		'format' => get_query_var( 'format_type' ) ?? '',
+		'month'  => get_query_var( 'month' ) ?? '',
+	);
+	array_walk( $facets, 'sanitize_text_field' );
+	$facets = array_filter( $facets );
+
+	$events = get_events( $attributes['events'], 0, 0, $facets );
+
+	// Get all the filters that are currently applied.
+	$filtered_events = array_slice( filter_events( $events ), 0, 10);
```

Comment:
This is cutting the list off at 10, but we only want that for the homepage. Maybe the block should accept a `limit` attribute that defaults to `50` or something?

Reply:
Yep, good idea. I was thinking about the grouping as well. Should we pass in a `groupyByMonth` property? Feels too specific.

Alternatively, we can pass a "heading" property to this list block and create a wrapper block where we use the query results to loop and output multiple instances of the event list block. I'll look at that today.